src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,581 @@
+    }
+
+    /**
+     * Change ZR camera mute status
+     *
+     * @param status true or false indicating the mute status
+     * @throws Exception during ssh communication
+     */
+    private void switchCameraMuteStatus(boolean status) throws Exception {
+        String meetingStatus = getCallStatus();
+        if (!meetingStatus.equals(IN_MEETING)) {
+            if(logger.isDebugEnabled()){
+                logger.debug("Not in a meeting. Not able to change camera mute status.");
+            }
+            return;
+        }
+        String command = status ? ON : OFF;
+        executeAndVerify(String.format(ZCOMMAND_CAMERA_MUTE, command), "*c zConfiguration Call Camera Mute");
+    }
+
+    /**
+     * Disconnect from the active call or from an idle call (ZR is trying to connect)
+     *
+     * @return true if disconnected successfully, false if an error has occured
+     */
+    private boolean callDisconnect() throws Exception {
+        meetingNumber = "";
+        if (isHost) {
+            return executeAndVerify(ZCOMMAND_CALL_DISCONNECT, "*r CallDisconnectResult");
+        } else {
+            return executeAndVerify(ZCOMMAND_CALL_LEAVE, "*r CallDisconnectResult");
+        }
+    }
+
+    /**
+     * Execute command using execute() method and simply check whether the command completed successfully
+     *
+     * @param command      to perform
+     * @param verifyString string that should be present in a response payload
+     * @return boolean value indicating whether the operation is successful
+     */
+    private boolean executeAndVerify(String command, String verifyString) throws Exception {
+        String response = execute(command, verifyString);
+        return !StringUtils.isNullOrEmpty(response);
+    }
+
+    /**
+     * Execute the command and check the response to contain "verifyString" in a few attempts.
+     * It's not added into the general "CommandSuccess/Failure" list because the basic communicator
+     * checks for endsWith() but there's no guarantee that the response will end with a certain string,
+     * also sometimes there are multiple comands perfomed automatically on entry with multiple "** end" breaks.
+     * This part helps to have another layer of response verification.
+     *
+     * @param command      to perform
+     * @param verifyString string that should be present in a response payload
+     * @return empty string or a command response, if successful
+     */
+    private String execute(String command, String verifyString) throws Exception {
+        String response;
+        controlOperationsLock.lock();
+        try {
+            refreshSshConnection();
+            response = send(command);
+            int retryAttempts = 0;
+            while (!response.contains(verifyString)) {
+                retryAttempts++;
+                response = send(command);
+                if (retryAttempts >= 10) {

```

Comment:
Why do we need to re-try here? And if we need, why that many times?

Reply:
Depending on the current state of the connection, Zoom Rooms CLI may throw a lot of unnecessary data which will not let the command to succeed. It happens generally on a re-login, but there's quite a lot of data, multiple screens, data like phonebook, for instance. Our account have about 150 entries, and each entry has detailed information, all that must be skipped in order to get the command through. Sure I would try other suggestions here but this one's been stable so far.

Reply:
Sorry but I do not understand what do you mean. If you are saying that there are irrelevant data on a wire which do not allow to parse response properly, why do we need to re-send actual command again and again? If it is something else, then I do not understand, please explain.
According to this code, we are going to re-send the the same command (e.g. hangup) up to 10 times. I cannot agree with such code unless I have a clear understanding why. If I do exactly the same from SSH session manually, I'm sure I do not need to type the same command and hit enter 10 times.

Reply:
The response like that would be splitted onto chunks, based on the buffer lengt, any command sent will just trigger next chunk to be delivered, using the default ssh client will make this process seamless, so the entire payload will be populated all at once. 
Currently, having the specific "verifyString" value passed to the method, will guarantee that the response is addressed for this specific command - whether it was successful or not. Another way of doing this would be sending the return character, until the return character is received as a response, but functionally it would be the same since the command is ignored, unless cli is ready to process it. 
I can change the approach accordingly, or change the buffer size (which, as I understand, would be a less appropriate solution)

Reply:
OK I think I understand it. It is not about buffer size, it is about a logic determining when to stop reading (implemented in doneReading method of ShellCommunicator). It uses technique to look for certain lines in the data read and determine that it is full response or not. Looks like your code stops reading prematurely, that's why you need to send something on the wire to start reading again. I really think you should investigate more if you can use this technique. I can help if you send me examples of commands you send and responses you received, especially in the case of when you have extra data there which you need to skip/ignore.
And if it is not possible to use that technique - then you should refactor your approach and:
- ideally try reading without sending new command, but while read method is accessible (it is protected), the input stream is not, so it will require changes in parent classes which for now is not an option
- at least send a \n instead of real command
- document it properly so it is clear what is it doing and why.

Reply:
Thank you! Will check a few things there and try refactoring.
As for the doneReading() method - yes, I did think about this and basically premature end of reading happens because there's no "real" generic end of the output. For instance - normally, the command response would end with "** end" and communicator is set up to process that, however, during the automatic output that I have mentioned - more than one such output may appear throughout the response, which will prematurely end reading and basically prevent us from knowing exactly whether this is a real end of an input or not. So when the real command is sent - with current solution it will be ignored if the output is not complete, and it will be processed and validated otherwise. 

In any case, I will go through the scenario once more when I have an access to ZR back to being able to refactor or provide additional details if it won't work.

Reply:
So? I do not see any changes here. Sorry but you will have a hard time to convince me that it is ok to resend exactly the same command 10 times just because of the fact that you cannot read full response. This does not pass real-life test.